### DISCUSSION TITLE
Is it possible to access other properties from a custom rule?

### DISCUSSION THREAD

**Discussion by basepack** [filamentphp/filament#1029] [Help] [answered]
I want to create a custom rule which needs to access another property in the form.

Form:
```php
             TextInput::make('someProperty')
                    ->rules([
                        function () {
                            return function (string $attribute, $value, \Closure $fail) {
                                if ($value === $this->otherProperty) {    // <--- undefined
                                    $fail("The {$attribute} is invalid.");
                                }
                            };
                        },]),
```

When I do it this way, otherProperty is undefined. Any thoughts?

**Comment by danharrin** [comment, marked as answer]
Try this:

```php
             TextInput::make('someProperty')
                    ->rules([
                        function ($get) {
                            return function (string $attribute, $value, \Closure $fail) {
                                if ($value === $get('anotherProperty')) {
                                    $fail("The {$attribute} is invalid.");
                                }
                            };
                        },]),
```

See https://filamentadmin.com/docs/2.x/forms/advanced#using-closure-customisation.

Also - Laravel / Filament has a dedicated rule for this already :)

```php
->same('anotherProperty')
```

**Reply by basepack** [reply]
Thanks Dan, great answer, fixed all my troubles :) I would not like to do a same statement, I left my real use case out for an easier question, but thanks for the suggestion. 

Btw. that docs link is great, haven't seen that one yet, a very good read.

**Reply by ibnnajjaar** [reply]
Does this solution work for others? Cox I am getting an error saying "Undefined variable $get" .

**Reply by ibnnajjaar** [reply]
Though this method doesn't work for me, I found a way to fix it. Check the comments here
https://github.com/filamentphp/filament/discussions/3868

Thanks.
